docs/standard/serialization/binaryformatter-migration-guide/wpf-applications.md

```diff
@@ -0,0 +1,92 @@
+---
+title: "BinaryFormatter migration guide: WPF applications"
+description: "Learn about the effects of the deprecation and removal of BinaryFormatter from .NET on WPF and how to migrate."
+ms.date: 8/05/2024
+no-loc: [BinaryFormatter, WPF]
+helpviewer_keywords:
+  - "BinaryFormatter"
+  - "WPF"
+---
+
+# WPF migration guide for BinaryFormatter
+
+## BinaryFormatter removal
+
+Starting with .NET 9, `BinaryFormatter` is no longer supported due to its known [security risks](../binaryformatter-security-guide.md) and its APIs always throw an exception for all project types, including WPF apps. For more information about the risks BinaryFormatter poses and the decision on its removal, see the [BinaryFormatter migration guide](index.md).
+
+With BinaryFormatter’s removal, it's expected that many WPF applications will be impacted, and you'll need to take action to complete your migration to .NET 9 or a later version.
+
+## How BinaryFormatter affects WPF
+
+Prior to .NET 9, Windows Presentation Foundation (WPF) used `BinaryFormatter` to serialize and deserialize data for scenarios such clipboard, drag-and-drop, and to store data for Avalon Binding in Journal. Starting with .NET 9, WPF and Windows Forms use a subset of the `BinaryFormatter` implementation internally for these scenarios. While BinaryFormatter's risks cannot be addressed in general-purpose serialization/deserialization, measures have been taken to mitigate the risks in these very specific use cases with a known set of types. A fall-back to `BinaryFormatter` is still in place for unknown or unsupported types, which will throw exceptions unless migration steps are taken in the application.
```

Comment:
I'm unfamiliar with Avalon Binding in Journal. Is 'Avalon' a well-known part of this feature name and/or is there a doc we should link to for this?

Reply:
Many people know that Avalon was the codename for WPF (that's why there is Avalonia) but it's not used in our docs. Unless we can point to something specific, I'm in favor of deleting the reference to 'Avalon Binding in Journal'.

Reply:
Thanks for pointing it out.
I wanted to convey "Holds the data for Avalon BindProducts in the journal",  
I have updated it.